velox/exec/WriteProtocol.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::connector {
+class ConnectorInsertTableHandle;
+class ConnectorQueryCtx;
+} // namespace facebook::velox::connector
+
+namespace facebook::velox::exec {
+
+class Operator;
+
+enum class CommitStrategy { kNoCommit, kTaskCommit };
+
+class WriteParameters {
+ public:
+  virtual ~WriteParameters() = default;
+};
+
+class HiveWriteParameters : public WriteParameters {
```

Comment:
WriteProtocol is suppose to be some abstract interface.
Yet here we have HiveWriteParameters.
I believe it should live in connectors/hive